File: Belajar/ProgramATM_SwitchCase/SwitchATM.py
while True:
    nama = input("Masukkan nama anda: ")
    konfirmasi = ""

    while konfirmasi not in ["true", "false"]:
        konfirmasi = input("Apakah nama sudah benar? (true/false): ").lower()

    if konfirmasi == "true":
        break
    print("Silakan masukkan nama kembali.\n")

while True:
    nim = input("Masukkan NIM: ")
    if nim.isdigit():
        break
    print("NIM harus angka!\n")

saldo = int(nim)

print(f"\nNama: {nama}")
print(f"Saldo Awal: Rp {saldo:,}\n".replace(",", "."))

while True:
    print("===== MENU ATM =====")
    print("1. Cek Saldo")
    print("2. Tarik Tunai")
    print("3. Setor Tunai")
    print("4. Transfer")
    print("5. Keluar")

    pilih = input("Pilih menu (1-5): ")
    print()

    if pilih == "1":
        print(f"Saldo: Rp {saldo:,}\n".replace(",", "."))

    elif pilih == "2":
        tarik = input("Masukkan jumlah tarik tunai: ")
        if tarik.isdigit() and int(tarik) > 0 and int(tarik) <= saldo:
            saldo -= int(tarik)
            print(f"Saldo baru: Rp {saldo:,}\n".replace(",", "."))
        else:
            print("Input tidak valid atau saldo kurang!\n")

    elif pilih == "3":
        setor = input("Masukkan jumlah setor: ")
        if setor.isdigit() and int(setor) > 0:
            saldo += int(setor)
            print(f"Saldo baru: Rp {saldo:,}\n".replace(",", "."))
        else:
            print("Input tidak valid!\n")

    elif pilih == "4":
        transfer = input("Masukkan nominal transfer: ")
        if transfer.isdigit() and int(transfer) > 0 and int(transfer) <= saldo:
            rek = input("Masukkan nomor rekening tujuan: ")
            saldo -= int(transfer)
            print(f"Transfer berhasil ke {rek}")
            print(f"Saldo baru: Rp {saldo:,}\n".replace(",", "."))
        else:
            print("Input tidak valid atau saldo kurang!\n")

    elif pilih == "5":
        print("Terima kasih!")
        break

    else:
        print("Pilihan tidak valid!\n")

File: Belajar/ProgramATM_SwitchCase/SwitchATM.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

int main() {
    string nama;
    string konfirmasi;

    while (true) {
        cout << "Masukkan nama anda: ";
        getline(cin, nama);

        konfirmasi = "";
        while (konfirmasi != "true" && konfirmasi != "false") {
            cout << "Apakah nama sudah benar? (true/false): ";
            getline(cin, konfirmasi);
        }

        if (konfirmasi == "true") break;

        cout << "Silakan masukkan nama kembali.\n";
    }

    string nim;
    bool nimBenar = false;

    while (!nimBenar) {
        cout << "Masukkan NIM: ";
        getline(cin, nim);

        bool angka = true;
        for (char c : nim) {
            if (!isdigit((unsigned char)c)) angka = false;
        }

        if (angka && nim != "") {
            nimBenar = true;
        } else {
            cout << "NIM harus angka!\n";
        }
    }

    long long saldo = stoll(nim);

    cout << "\nNama: " << nama << "\n";
    cout << "Saldo Awal: Rp " << saldo << "\n\n";

    while (true) {
        cout << "===== MENU ATM =====\n";
        cout << "1. Cek Saldo\n";
        cout << "2. Tarik Tunai\n";
        cout << "3. Setor Tunai\n";
        cout << "4. Transfer\n";
        cout << "5. Keluar\n";
        cout << "Pilih menu (1-5): ";

        string pilih;
        getline(cin, pilih);
        cout << "\n";

        if (pilih == "1") {
            cout << "Saldo: Rp " << saldo << "\n\n";
        }

        else if (pilih == "2") {
            string tarik;
            long long nominal = -1;

            while (nominal <= 0) {
                cout << "Masukkan jumlah tarik tunai: ";
                getline(cin, tarik);

                bool angka = true;
                for (char c : tarik) {
                    if (!isdigit((unsigned char)c)) angka = false;
                }

                if (!angka || tarik == "") {
                    cout << "Input harus angka dan lebih dari 0!\n";
                } else {
                    nominal = stoll(tarik);
                }
            }

            if (nominal > saldo) {
                cout << "Saldo tidak cukup!\n\n";
            } else {
                saldo -= nominal;
                cout << "Saldo baru: Rp " << saldo << "\n\n";
            }
        }

        else if (pilih == "3") {
            string setor;
            long long nominal = -1;

            while (nominal <= 0) {
                cout << "Masukkan jumlah setor: ";
                getline(cin, setor);

                bool angka = true;
                for (char c : setor) {
                    if (!isdigit((unsigned char)c)) angka = false;
                }

                if (!angka || setor == "") {
                    cout << "Input harus angka dan lebih dari 0!\n";
                } else {
                    nominal = stoll(setor);
                }
            }

            saldo += nominal;
            cout << "Saldo baru: Rp " << saldo << "\n\n";
        }

        else if (pilih == "4") {
            string transfer;
            long long nominal = -1;

            while (nominal <= 0) {
                cout << "Masukkan nominal transfer: ";
                getline(cin, transfer);

                bool angka = true;
                for (char c : transfer) {
                    if (!isdigit((unsigned char)c)) angka = false;
                }

                if (!angka || transfer == "") {
                    cout << "Input harus angka dan lebih dari 0!\n";
                } else {
                    nominal = stoll(transfer);
                }
            }

            if (nominal > saldo) {
                cout << "Saldo tidak cukup!\n\n";
            } else {
                string rek;
                cout << "Masukkan nomor rekening tujuan: ";
                getline(cin, rek);

                saldo -= nominal;
                cout << "Transfer berhasil ke " << rek << "\n";
                cout << "Saldo baru: Rp " << saldo << "\n\n";
            }
        }

        else if (pilih == "5") {
            cout << "Terima kasih!\n";
            return 0;
        }

        else {
            cout << "Pilihan tidak valid!\n\n";
        }
    }
}
